### PR TITLE
test: fix unstable tests using failpoint

### PR DESCRIPTION
**Pull request by wjhuang2016**
Signed-off-by: wjhuang2016 <huangwenjun1997@gmail.com>

### What problem does this PR solve?

`go test -v -vet=off -p 20 -timeout 10m github.com/pingcap/tidb/executor`
There are so many fail tests!!!
The tests use failpoint, but these tests are not serial. 

### Check List 

Tests 

- No code

Side effects

### Release note 

- No release note

### CODE REVIEW

File: executor/executor_test.go [hunk 1]
```diff
@@ -2340,7 +2342,7 @@ func (s *testSuiteP2) TestBatchPointGetRepeatableRead(c *C) {
 	c.Assert(failpoint.Disable(step2), IsNil)
 }
 
-func (s *testSuite7) TestSplitRegionTimeout(c *C) {
+func (s *testSerialSuite) TestSplitRegionTimeout(c *C) {
 	c.Assert(failpoint.Enable("github.com/pingcap/tidb/store/tikv/MockSplitRegionTimeout", `return(true)`), IsNil)
```
Comment (bb7133) [inline executor/executor_test.go:2346]: So other test cases may be failed because of the failpoint here?

Comment (wjhuang2016) [reply executor/executor_test.go:2346]: Maybe, anyway, it's not safe.

**Review by winoros — COMMENTED** [review]
LGTM

**Comment by lance6716** [comment]
BTW, `go test -v -vet=off -p 20 -timeout 10m -race github.com/pingcap/tidb/planner/core -check.p` seems have same problem?

https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/48303/pipeline/73/

**Comment by wjhuang2016** [comment]
> BTW, `go test -v -vet=off -p 20 -timeout 10m -race github.com/pingcap/tidb/planner/core -check.p` seems have same problem?
> 
> https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/48303/pipeline/73/

But I can't reproduce it in my local machine.

**Comment by bb7133** [comment]
LGTM

**Review by winoros — APPROVED** [review]

**Pull request merged**